docs/content/sips/0xx-component-dependencies.md

```diff
@@ -0,0 +1,108 @@
+
+# Optionally, explicitly specify the export name to use to satisfy the `aws:client/s3` interface.
+"aws:client/s3" = { path = "my_aws_client.wasm", export = "my-s3-client" }
+
+# Without an interface name, attempt to resolve every import of the "aws:client" package.
+"aws:client" = { path = "my_aws_client.wasm" }
+```
+
+> ⚠️ NOTE: Using the package and interface forms together (e.g. `aws:client` and `aws:client/s3` respectively) within the same dependencies section will result in an error to prevent resolution ambiguities.
+
+## Using dependencies from a registry (remote)
+```toml
+[component."infra-dashboard".dependencies]
+# Equivalent to { version = "1.0.0" , package = "aws:client"}
+"aws:client" = "1.0.0"
+
+# Use the `aws:client` component package to satisfy any number of "wasi:blobstore" imports ...
+"wasi:blobstore" = { registry = "my-registry.io", version = "0.1.0", package = "aws:client" }
+```
+
+## Dependency Isolation
+Statically composing the parent Spin component and its dependencies yields a component where the parent component and dependencies are no longer isolated from eachother. Each dependency will inherit the configuration of the parent component, which may be desireable in some library use-cases (e.g., allowing the `aws:client/s3` dependency to make outbound requests via `allowed_outbound_hosts`). However, making this the default behavior contradicts the explicit opt-in nature of capabilities for Spin (and the component model), so by default, component dependencies in Spin will not inherit configuration.
+
+To allow dependencies to inherit configuration from the parent Spin component, developers can set `dependencies_inherit_configuration = true`, as shown in the following example:
+
+```toml
+[component."infra-dashboard"]
+# ...
+allowed_outbound_hosts = ["https://s3.us-west-2.amazonaws.com/my-bucket/puppy.jpeg"]
+dependencies_inherit_configuration = true
+
+[component."infra-dashboard".dependencies]
+"aws:client" = "1.0.0"
+```
+
+The above configuration allows the component used to satisfy the `aws:client` dependency to make outbound HTTP requests to the S3 bucket specified in `allowed_outbound_hosts`. If `dependencies_inherit_configuration` is omitted or set to `false`, an `AccessDenied` error will be returned whenever the `aws:client` component attempts to make an outbound HTTP request.
+
+### The `spin-deny-all` adapter
+To implement the default behavior (i.e. `dependencies_inherit_configuration = false`), Spin will apply what's called the `spin-deny-all` adapter to maintain isolation between components. When configuration inheritance is disabled, Spin will compose the ahead-of-time constructed deny adapter with each dependency, resulting in a new component that denies access to various capabilities (e.g. `wasi:http/outgoing-handler`, `fermyon:spin/variables`, etc.).
+
+The `spin-deny-all` adapter is a component that targets the following world:
+
+```
+package fermyon:spin-virt;
+
+world deny-all {
+    # WASI
```

Comment:
I'm not sure this is going to be possible without virtualizing all of wasi (due to the `wasi:io` issue we've talked above before). I guess that's fine, since we'll need to do this in the future any way if we want to support component isolation, but it does significantly complicate the implementation.

Reply:
Given the context Lann provided above and since we discussed out-of-band i'll resolve this thread! 

tl;dr for anyone skimming: because we aren't actually virtualizing `wasi:io` and simply passing through we avoid all the complicate details that Ryan is referring to.